ionic_langchain/tool.py

```diff
@@ -18,20 +20,53 @@ def __init__(self, sdk: Optional[IonicSDK] = None):
         else:
             self._sdk = IonicSDK()
 
-    def query(self, queries: str) -> list:
-        """
-        FIXME: handle non-200 responses
-        TODO: better typing in response
-        """
+    def query(
+        self,
+        query_input: str,
+    ) -> Sequence[dict[str, Any]]:
+        if not query_input:
+            raise ValueError("query must not be empty")
+
+        query, num_results, min_price, max_price = self.gen_query_request(query_input)
         request = QueryAPIRequest(
-            queries=[Query(query=query) for query in queries.split(", ")],
+            query=SDKQuery(
+                query=query,
+                num_results=num_results,
+                min_price=min_price,
+                max_price=max_price,
+            )
         )
         response: QueryResponse = self._sdk.query(
             request=request,
             security=QuerySecurity(),
         )
 
-        return [dataclasses.asdict(r) for r in response.query_api_response.results]
+        return [
+            dataclasses.asdict(result) for result in response.query_api_response.results
+        ]
+
+    def _parse_number(self, value: str) -> int | None:
+        return int(value) if value and int(value) >= 0 else None
+
+    def gen_query_request(self, query_input: str) -> tuple[str, int | None, int | None, int | None]:
+        if not query_input:
+            raise ValueError("query must not be empty")
+
+        split_query = query_input.split(",")
+        len4_query = split_query + [None] * (4 - len(split_query))  # pad with None
+
+        query, num_results, min_price, max_price, *rest = [  # *rest ignores extra values
```

Comment:
How certain are we that the given `query_input` matches the correct format?

Reply:
its the format that is specified in the prompt. 😅